*** ../sxiv2/commands.c	2020-11-17 14:39:57.872806313 -0500
--- ../sxiv/commands.c	2020-11-17 13:49:37.608702983 -0500
*************** bool cg_navigate_marked(arg_t n)
*** 240,246 ****
  	if (prefix > 0)
  		n *= prefix;
  	d = n > 0 ? 1 : -1;
! 	for (i = fileidx + d; n != 0 && i >= 0 && i < filecnt; i += d) {
  		if (files[i].flags & FF_MARK) {
  			n -= d;
  			new = i;
--- 240,250 ----
  	if (prefix > 0)
  		n *= prefix;
  	d = n > 0 ? 1 : -1;
! 	for (i = fileidx + d; n != 0; i += d) {
! 		if (i < 0)
! 			i += filecnt;
! 		else if (i >= filecnt)
! 			i -= filecnt;
  		if (files[i].flags & FF_MARK) {
  			n -= d;
  			new = i;
*************** bool ci_navigate(arg_t n)
*** 275,284 ****
  	if (prefix > 0)
  		n *= prefix;
  	n += fileidx;
! 	if (n < 0)
! 		n = 0;
! 	if (n >= filecnt)
! 		n = filecnt - 1;
  
  	if (n != fileidx) {
  		load_image(n);
--- 279,288 ----
  	if (prefix > 0)
  		n *= prefix;
  	n += fileidx;
! 	while (n < 0)
! 		n += filecnt;
! 	while (n >= filecnt)
! 		n -= filecnt;
  
  	if (n != fileidx) {
  		load_image(n);
