*** ../sxiv2/thumbs.c	2020-11-17 15:21:03.389890665 -0500
--- ../sxiv/thumbs.c	2020-11-17 15:18:13.359884847 -0500
*************** bool tns_move_selection(tns_t *tns, dire
*** 510,527 ****
  
  	switch (dir) {
  		case DIR_UP:
! 			*tns->sel = MAX(*tns->sel - cnt * tns->cols, *tns->sel % tns->cols);
  			break;
  		case DIR_DOWN:
! 			max = tns->cols * ((*tns->cnt - 1) / tns->cols) +
! 			      MIN((*tns->cnt - 1) % tns->cols, *tns->sel % tns->cols);
! 			*tns->sel = MIN(*tns->sel + cnt * tns->cols, max);
  			break;
  		case DIR_LEFT:
! 			*tns->sel = MAX(*tns->sel - cnt, 0);
  			break;
  		case DIR_RIGHT:
! 			*tns->sel = MIN(*tns->sel + cnt, *tns->cnt - 1);
  			break;
  	}
  
--- 510,537 ----
  
  	switch (dir) {
  		case DIR_UP:
! 			max = *tns->cnt / tns->cols +
! 				(*tns->sel % tns->cols < *tns->cnt % tns->cols);
! 			*tns->sel -= cnt * tns->cols;
! 			while (*tns->sel < 0)
! 				*tns->sel += max * tns->cols;
  			break;
  		case DIR_DOWN:
! 			max = *tns->cnt / tns->cols +
! 				(*tns->sel % tns->cols < *tns->cnt % tns->cols);
! 			*tns->sel += cnt * tns->cols;
! 			while (*tns->sel >= *tns->cnt)
! 				*tns->sel -= max * tns->cols;
  			break;
  		case DIR_LEFT:
! 			*tns->sel -= cnt;
! 			while (*tns->sel < 0)
! 				*tns->sel += *tns->cnt;
  			break;
  		case DIR_RIGHT:
! 			*tns->sel += cnt;
! 			while (*tns->sel >= *tns->cnt)
! 				*tns->sel -= *tns->cnt;
  			break;
  	}
  
